.github/workflows/ruby.yml

```diff
@@ -8,7 +8,7 @@ jobs:
 
     strategy:
       matrix:
-        ruby-version: ['2.4.0', '2.5.1', '2.6.6', '2.7.2', 'ruby-head']
+        ruby-version: ['2.6.10', '2.7.8', '3.0.6', '3.1.4', '3.2.2', 'ruby-head']
```

Comment:
By removing these versions from the matrix, we are effectively dropping support for them. 

My thinking is this means we need to bump versions and update the changelog. I'm not well-versed enough with semver etiquette to say whether this would be a minor or a major version bump. There are no breaking functional changes, but we are dropping explicit support, so future breaking changes could be added, and we wouldn't catch them in the test suite.

Reply:
Yup! We plan to release a major version bump, so folks can still use the old one if they want.

Reply:
@roland-at-handshake @amy-hs Updated the changelog and version. Once this is merged, I will run the release script to push to rubygems. 

I try to follow this document on how to use semver and when to bump what, https://semver.org/. I believe this warrants a major bump but open to other thoughts.